aries/aries_vcx/src/handlers/out_of_band/receiver.rs

```diff
@@ -62,11 +84,45 @@ impl OutOfBandReceiver {
         self.oob.clone().into()
     }
 
-    pub fn from_string(oob_data: &str) -> VcxResult<Self> {
-        Ok(Self {
-            oob: serde_json::from_str(oob_data)?,
-        })
+    pub fn to_json_string(&self) -> String {
+        self.to_aries_message().to_string()
     }
+
+    pub fn to_base64_url(&self) -> String {
+        BASE64_URL_SAFE.encode(self.to_json_string())
+    }
+
+    pub fn to_url(&self, domain_path: &str) -> VcxResult<Url> {
+        let mut oob_url = Url::parse(domain_path)?;
+        let oob_query = "oob=".to_owned() + &self.to_base64_url();
+        oob_url.set_query(Some(&oob_query));
+        Ok(oob_url)
+    }
+}
+
+fn from_json_string(oob_json: &str) -> VcxResult<Invitation> {
+    Ok(serde_json::from_str(oob_json)?)
+}
+
+fn from_base64_url(base64_url_encoded_oob: &str) -> VcxResult<String> {
+    Ok(String::from_utf8(
+        URL_SAFE_LENIENT.decode(base64_url_encoded_oob)?,
```

Comment:
nice using the lenient engine. i was getting ready to nitpick it if you hadn't ;)